aea/crypto/cosmos.py

```diff
@@ -78,6 +78,38 @@ def is_transaction_settled(tx_receipt: JSONLike) -> bool:
                 )
         return is_successful
 
+    @staticmethod
+    def get_code_id(tx_receipt: JSONLike) -> Optional[int]:
+        """
+        Retrieve the `code_id` from a transaction receipt.
+
+        :param tx_receipt: the receipt of the transaction.
+        :return: the code id, if present
+        """
+        code_id: Optional[int] = None
+        try:
+            res = [dic_["value"] for dic_ in tx_receipt["logs"][0]["events"][0]["attributes"] if dic_["key"] == "code_id"]  # type: ignore
+            code_id = int(res[0])
+        except (KeyError, IndexError):
+            code_id = None
+        return code_id
+
+    @staticmethod
+    def get_contract_address(tx_receipt: JSONLike) -> Optional[str]:
+        """
+        Retrieve the `contract_address` from a transaction receipt.
+
+        :param tx_receipt: the receipt of the transaction.
+        :return: the contract address, if present
+        """
+        contract_address: Optional[str] = None
+        try:
+            res = [dic_["value"] for dic_ in tx_receipt["logs"][0]["events"][0]["attributes"] if dic_["key"] == "contract_address"]  # type: ignore
+            contract_address = res[0]
+        except (KeyError, IndexError):
+            contract_address = None
+        return contract_address
+
```

Comment:
Why you don't return the code_id as a receipt, when we deploy and then contract address when we initialise the contract ? Similar to digest that you return the hash. Do we need to make an extra call to the API ? So both of these function can be private, to the API.

Reply:
These are helper methods to apply to the tx_receipt. So no further api call needed.